kinto/core/storage/postgresql/__init__.py

```diff
@@ -218,10 +218,12 @@ def create(self, collection_id, parent_id, record, id_generator=None,
                auth=None, ignore_conflict=False):
         id_generator = id_generator or self.id_generator
         record = {**record}
-        if id_field in record and not ignore_conflict:
+        if id_field in record:
             # Raise unicity error if record with same id already exists.
             try:
                 existing = self.get(collection_id, parent_id, record[id_field])
+                if ignore_conflict:
+                    return existing
```

Comment:
We should still remove the tombstone in that case.

What about:

```diff
diff --git a/kinto/core/storage/memory.py b/kinto/core/storage/memory.py
index f86f10d..0650955 100644
--- a/kinto/core/storage/memory.py
+++ b/kinto/core/storage/memory.py
@@ -135,22 +135,23 @@ class Storage(MemoryBasedStorage):
                modified_field=DEFAULT_MODIFIED_FIELD, auth=None, ignore_conflict=False):
         id_generator = id_generator or self.id_generator
         record = {**record}
         if id_field in record:
-            if not ignore_conflict:
-                # Raise unicity error if record with same id already exists.
-                try:
-                    existing = self.get(collection_id, parent_id, record[id_field])
-                    raise exceptions.UnicityError(id_field, existing)
-                except exceptions.RecordNotFoundError:
-                    pass
+            # Raise unicity error if record with same id already exists.
+            try:
+                record = self.get(collection_id, parent_id, record[id_field])
+                if not ignore_conflict:
+                    raise exceptions.UnicityError(id_field, record)
+            except exceptions.RecordNotFoundError:
+                pass
```


The storage tests on `ignore_conflict=True` should be more explicit regarding:
* removing the tombstone or not
* bumping the collection timestamp or not